pkg/runtime/framework/plugins/torch/torch_test.go

```diff
@@ -0,0 +1,516 @@
+	"github.com/kubeflow/trainer/pkg/runtime"
+	"github.com/kubeflow/trainer/pkg/runtime/framework"
+	utiltesting "github.com/kubeflow/trainer/pkg/util/testing"
+)
+
+func TestTorch(t *testing.T) {
+	cases := map[string]struct {
+		info               *runtime.Info
+		trainJob           *trainer.TrainJob
+		wantInfo           *runtime.Info
+		wantMLPolicyError  error
+		wantNumProcPerNode string // For validating numProcPerNode value
+	}{
+		"no action when info is nil": {},
+		"no action when mlPolicy is nil": {
+			info: runtime.NewInfo(
+				runtime.WithLabels(map[string]string{"key": "value"}),
+			),
+			wantInfo: runtime.NewInfo(
+				runtime.WithLabels(map[string]string{"key": "value"}),
+			),
+		},
+		"no action when mlPolicy torch is null": {
+			info: runtime.NewInfo(
+				runtime.WithMLPolicy(utiltesting.MakeMLPolicyWrapper().
+					Obj()),
+			),
+			wantInfo: runtime.NewInfo(
+				runtime.WithMLPolicy(utiltesting.MakeMLPolicyWrapper().
+					Obj()),
+			),
+		},
+		"trainJob numNodes is respected rather than mlPolicy one": {
+			info: runtime.NewInfo(
+				runtime.WithMLPolicy(
+					utiltesting.MakeMLPolicyWrapper().
+						WithNumNodes(1).
+						TorchPolicy("auto", nil).
+						Obj(),
+				),
+			),
+			trainJob: utiltesting.MakeTrainJobWrapper(metav1.NamespaceDefault, "trainJob").
+				Trainer(
+					utiltesting.MakeTrainJobTrainerWrapper().
+						NumNodes(2).
+						Obj()).
+				Obj(),
+			wantInfo: &runtime.Info{
+				Labels:      make(map[string]string),
+				Annotations: make(map[string]string),
+				RuntimePolicy: runtime.RuntimePolicy{
+					MLPolicy: utiltesting.MakeMLPolicyWrapper().
+						WithNumNodes(1).
+						TorchPolicy("auto", nil).
+						Obj(),
+				},
+				Trainer: runtime.Trainer{
+					NumNodes:       ptr.To[int32](2),
+					NumProcPerNode: "",
+					Env: []corev1ac.EnvVarApplyConfiguration{
+						{
+							Name:  ptr.To(constants.TorchEnvNumNodes),
+							Value: ptr.To("2"),
+						},
+						{
+							Name:  ptr.To(constants.TorchEnvNumProcPerNode),
+							Value: ptr.To("auto"),
+						},
+						{
+							Name: ptr.To(constants.TorchEnvNodeRank),
+							ValueFrom: &corev1ac.EnvVarSourceApplyConfiguration{
+								FieldRef: &corev1ac.ObjectFieldSelectorApplyConfiguration{
+									FieldPath: ptr.To(constants.JobCompletionIndexFieldPath),
+								},
+							},
+						},
+						{
+							Name:  ptr.To(constants.TorchEnvMasterAddr),
+							Value: ptr.To("trainJob-trainer-node-0-0.trainJob"),
+						},
+						{
+							Name:  ptr.To(constants.TorchEnvMasterPort),
+							Value: ptr.To(fmt.Sprintf("%d", constants.ContainerTrainerPort)),
+						},
+					},
+					ContainerPort: &corev1ac.ContainerPortApplyConfiguration{
+						ContainerPort: ptr.To[int32](constants.ContainerTrainerPort),
+					},
+				},
+				Scheduler: &runtime.Scheduler{TotalRequests: map[string]runtime.TotalResourceRequest{}},
+			},
+		},
+		// Issue #2407 test case - nproc_per_node=auto with CPU limit
```

Comment:
```suggestion
		// nproc_per_node=auto with CPU limit
```

It would be better not to add such a specific prefix.
Could you address those in other places, as well?